lib/graphql/execution/lookahead.rb

```diff
@@ -94,6 +94,43 @@ def selection(field_name, arguments: nil)
         end
       end
 
+      # Like {#selection}, but for all nodes.
+      # It returns a list of Lookaheads for all Selections
+      #
+      # If `arguments:` is provided, each provided key/value will be matched
+      # against the arguments in each selection. This method will filter the selections
+      # if any of the given `arguments:` do not match the given selection.
+      #
+      # @example getting the name of a selection
+      #   def articles(lookahead:)
+      #     next_lookaheads = lookahead.selections # => [#<GraphQL::Execution::Lookahead ...>, ...]
+      #     next_lookaheads.map(&:name) #=> [:full_content, :title]
+      #   end
+      #
+      # @param arguments [Hash] Arguments which must match in the selection
+      # @return [Array<GraphQL::Execution::Lookahead>]
+      def selections(arguments: nil)
+        @ast_nodes
+          .flat_map(&:selections)
+          .map(&:name)
+          .map { |name| selection(name, arguments: arguments) }
+          .select(&:selected?)
+      end
+
+      # The method name of the field.
+      # It returns the method_sym of the Lookahead's field.
+      #
+      # @example getting the name of a selection
+      #   def articles(lookahead:)
+      #     article.selection(:full_content).name # => :full_content
+      #     # ...
+      #   end
+      #
+      # @return [Symbol]
+      def name
+        @field&.method_sym
```

Comment:
A couple thoughts here: 

- As you noticed on the CI failures, we support some rubies that don't have `&.`,  so this will have to be checked "manually", eg `@field && @field.something` 
- I don't think `.method_sym` is really what we want, because if the field is created with `field :name, ..., method: :some_other_method`, this will return `:some_other_method` instead of `:name`. As you probably noticed, there's _no way_ to get the originally-configured name 🤦‍♂️ . Maybe the thing to do is improve `schema/field.rb` here: 

  https://github.com/rmosolgo/graphql-ruby/blob/5527179a7b2f100ce9ee093764c924e87d155f1b/lib/graphql/schema/field.rb#L145

  To instead be like 
   ``` 
    @original_name = name 
    @name = # ... 
    ```

    And add an `original_name` reader method. Do you think that would do the trick?

Reply:
Good catch, did not realize that is what method_sym was. I will make that enhancement :smile: